compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1271,11 +1271,36 @@ LogicalResult AttentionOp::verify() {
   if (failed(checkShape("Query", getQueryType().getShape(), getQueryMap())) ||
       failed(checkShape("Key", getKeyType().getShape(), getKeyMap())) ||
       failed(checkShape("Value", getValueType().getShape(), getValueMap())) ||
+      (getMaskMap().has_value() &&
+       failed(checkShape("Mask", getMaskType().value().getShape(),
+                         getMaskMap().value()))) ||
       failed(
           checkShape("Output", getOutputType().getShape(), getOutputMap()))) {
     return failure();
   }
 
+  int expectedSymbols = getQueryMap().getNumInputs();
+  auto checkDomain =
+      [&attnOp, &expectedSymbols](StringRef operandName,
+                                  AffineMap indexingMap) -> LogicalResult {
+    if (expectedSymbols != indexingMap.getNumInputs()) {
+      return attnOp->emitError("Mismatched map domain for ")
+             << operandName << ". Expected: " << expectedSymbols
+             << " Got: " << indexingMap.getNumInputs();
+    }
+    return success();
+  };
+
+  if (failed(checkDomain("Query", getQueryMap())) ||
+      failed(checkDomain("Key", getKeyMap())) ||
+      failed(checkDomain("Value", getValueMap())) ||
+      failed(checkDomain("Scale", getScaleMap())) ||
+      (getMaskMap().has_value() &&
+       failed(checkDomain("Mask", getMaskMap().value()))) ||
+      failed(checkDomain("Output", getOutputMap()))) {
+    return failure();
```

Comment:
Could you move it to a separate check? This should make it more debuggable / readable.

Reply:
IE a separate if condition

Reply:
Sounds good